Assembler.Base/MessagesAssembler.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using Assembler.Core;
+using Assembler.Core.Entities;
+using Assembler.Core.Enums;
+
+namespace Assembler.Base
+{
+    public class MessagesAssembler : IAssembler
+    {
+        private readonly IResolver<FrameType, IHandler<BaseFrame, BaseMessageInAssembly>> _resolver;
+        private readonly IConverter<BaseMessageInAssembly, BaseAssembledMessage> _converter;
+        private readonly ILogger _logger;
+
+        public event Action<BaseAssembledMessage> MessageAssembled;
+
+        public MessagesAssembler(IResolver<FrameType, IHandler<BaseFrame, BaseMessageInAssembly>> resolver,
+            ITimeBasedCache<BaseMessageInAssembly> cache,
+            IEnumerable<IHandler<BaseFrame, BaseMessageInAssembly>> handlers, IConverter<BaseMessageInAssembly, BaseAssembledMessage> converter,
+            ILoggerFactory loggerFactory)
+        {
+            _resolver = resolver;
+            _converter = converter;
+            _logger = loggerFactory.GetLogger(this);
+
+            cache.ItemExpired += ReleaseExpiredMessage;
```

Comment:
Note that we'd usually have something like this in a `Start()` method. I don't completely object to this being in the c'tor but perhaps we would first like to rethink this as a convention

Reply:
I know, I'm pretty against the `Start()` method, lets see what other people think

Reply:
👍